### PR TITLE
[CAMEL-21362] camel k8s run cannot find pod on openshift

### PR DESCRIPTION
**Pull request by tdiesler**
https://issues.apache.org/jira/browse/CAMEL-21362

### CODE REVIEW

**Comment by davsclaus** [comment]
So it seems it needs 2 labels now.

File: dsl/camel-jbang/camel-jbang-plugin-kubernetes/src/main/java/org/apache/camel/dsl/jbang/core/commands/kubernetes/traits/BaseTrait.java [hunk 1]
```diff
@@ -20,6 +20,7 @@
 public abstract class BaseTrait implements Trait {
 
     public static final String INTEGRATION_LABEL = "camel.apache.org/integration";
```
Comment (squakez) [inline dsl/camel-jbang/camel-jbang-plugin-kubernetes/src/main/java/org/apache/camel/dsl/jbang/core/commands/kubernetes/traits/BaseTrait.java:22]: Probably you need to remove this then.

File: dsl/camel-jbang/camel-jbang-plugin-kubernetes/src/main/java/org/apache/camel/dsl/jbang/core/commands/kubernetes/traits/ContainerTrait.java [hunk 2]
```diff
@@ -82,6 +82,7 @@ public void apply(Traits traitConfig, TraitContext context) {
                 .editOrNewTemplate()
                 .editOrNewMetadata()
                 .addToLabels(INTEGRATION_LABEL, context.getName())
```
Comment (squakez) [inline dsl/camel-jbang/camel-jbang-plugin-kubernetes/src/main/java/org/apache/camel/dsl/jbang/core/commands/kubernetes/traits/ContainerTrait.java:84]: ditto

File: dsl/camel-jbang/camel-jbang-plugin-kubernetes/src/main/java/org/apache/camel/dsl/jbang/core/commands/kubernetes/traits/ContainerTrait.java [hunk 3]
```diff
@@ -93,6 +94,7 @@ public void apply(Traits traitConfig, TraitContext context) {
                 .editOrNewJobTemplate()
                 .editOrNewMetadata()
                 .addToLabels(INTEGRATION_LABEL, context.getName())
```
Comment (squakez) [inline dsl/camel-jbang/camel-jbang-plugin-kubernetes/src/main/java/org/apache/camel/dsl/jbang/core/commands/kubernetes/traits/ContainerTrait.java:96]: ditto

**Review by squakez — COMMENTED** [review]
I guess you may need to remove the Integration label at all which appears to be a Camel K leftover. Not sure if that would impact any other existing feature though.

**Comment by davsclaus** [comment]
thanks for the review @squakez

**Comment by tdiesler** [comment]
Should this be removed in 4.x as well? It'd be a breaking change.

**Comment by davsclaus** [comment]
> Should this be removed in 4.x as well? It'd be a breaking change.

yeah fine to remove in 4.8.x as its a brand new release and 4.8.0 is a bit buggy here

**Review by davsclaus — APPROVED** [review]

**Comment by gansheer** [comment]
LGTM.
A little out of scope but I saw this comment in the code : 
https://github.com/apache/camel/blob/f23f02531518c8a2b34a0b5d582a96e81fcd7c42/dsl/camel-jbang/camel-jbang-plugin-kubernetes/src/main/java/org/apache/camel/dsl/jbang/core/commands/kubernetes/KubernetesExport.java#L232-L237
Do you know if there is a lot of diff between jkube, quarkus and generated manifest for labels ?

**Comment by squakez** [comment]
> Should this be removed in 4.x as well? It'd be a breaking change.

It is a valid concern. However, as this is a prototyping tool, I think we should not worry that much as we don't expect production workflows leveraging this directly.

**Comment by christophd** [comment]
I think the root cause of all of this is that when using OpenShift as a `cluster-type` we switch from Kubernetes Quarkus extension to the OpenShift Quarkus extension and unfortunately these two seem to behave differently. 

The Quarkus Kubernetes extension properly merges the Kubernetes manifest coming from the Camel JBang plugin (in `src/main/kubernetes`) with its own generated Kubernetes manifest. The Quarkus OpenShift extension presumably is not doing the same merger so things like the INTEGRATION_LABEL may not get properly added to the final Kubernetes manifest.

Switching from INTEGRATION_LABEL to KUBERNETES_NAME_LABEL as in this PR fixes it because KUBERNETES_NAME_LABEL is what the Quarkus extension uses. I am fine with the fix in this PR and I also think that is does not break existing behavior since the users coming from 4.8.0 will not face breaking changes because of this as it is an internal detail in the manifest.

In general we need to analyze and find out why the Kubernetes manifest is not merged properly when using the Quarkus OpenShift extension. In 4.8.0 the focus was to support plain Kubernetes where all of this is working properly.

**Review by christophd — APPROVED** [review]

**Comment by davsclaus** [comment]
@christophd can you create a JIRA with your last comment so we wont forget

**Comment by christophd** [comment]
@davsclaus see https://issues.apache.org/jira/browse/CAMEL-21365

**Comment by davsclaus** [comment]
@tdiesler can you update the PR
	modified:   dsl/camel-jbang/camel-jbang-plugin-k/src/test/java/org/apache/camel/dsl/jbang/core/commands/k/IntegrationExportTest.java

**Pull request merged**